internal/controller/util/misc.go

```diff
@@ -254,3 +260,29 @@ func TrimToK8sResourceNameLength(name string) string {
 
 	return name
 }
+
+func GetJobName(namePrefix string, ownerName string) string {
+	return getShortenedResourceName(namePrefix, ownerName, JobNameMaxLength)
+}
+
+func GetServiceName(namePrefix string, ownerName string) string {
+	return getShortenedResourceName(namePrefix, ownerName, ServiceNameMaxLength)
+}
+
+func getShortenedResourceName(namePrefix string, ownerName string, maxLength int) string {
+	name := namePrefix + ownerName
+
+	if len(name) > maxLength {
+		// Use crc32 to hash the owner CR name
+		return namePrefix + GetHashedName(ownerName)
+	}
+
+	// No need to shorten, use original name
+	return name
+}
+
+// This mechanism is implemented in volsync operator
```

Comment:
Will useful to write here:

    // Implement the volsync shortening algorithm, required to match volsync resources names.

Reply:
Done